libnavigator/src/main/java/com/mapbox/navigation/navigator/internal/MapboxNativeNavigatorImpl.kt

```diff
@@ -188,31 +200,38 @@ object MapboxNativeNavigatorImpl : MapboxNativeNavigator {
      *
      * @param route [DirectionsRoute]
      */
-    override suspend fun updateAnnotations(route: DirectionsRoute): Unit =
-        withContext(NavigatorDispatcher) {
-            route.legs()?.forEachIndexed { index, routeLeg ->
+    override suspend fun updateAnnotations(route: DirectionsRoute) {
+        route.legs()?.forEachIndexed { index, routeLeg ->
```

Comment:
Can we update annotations for each leg in parallel? You can use `launch` or `async` for that.

Reply:
it's executed in a single thread on NN side, no benefits to run it in parallel